packages/contracts-bedrock/src/l1/SuperchainConfig.sol

```diff
@@ -0,0 +1,247 @@
+
+    /// @notice The address of the guardian, can pause the OptimismPortal.
+    ///         It can only be modified by an upgrade.
+    bytes32 public constant GUARDIAN_SLOT = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+
+    /// @notice The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    ///         It can only be modified by an upgrade.
+    bytes32 public constant DELAY_SLOT = bytes32(uint256(keccak256("superchainConfig.delay")) - 1);
+
+    /// @notice The pause status of withdrawals from an chain in the superchain.
+    ///         Set by the guardian role.
+    bytes32 public constant PAUSED_SLOT = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
+
+    /// @notice Mapping of allowed sequencers.
+    ///         The initiator should be able to add to it instantly, but removing is subject to delay.
+    mapping(bytes32 => bool) public allowedSequencers;
+
+    /// @notice Emitted when the pause is triggered.
+    event Paused();
+
+    /// @notice Emitted when the pause is lifted.
+    event Unpaused();
+
+    /// @notice Emitted when configuration is updated.
+    /// @param updateType Type of update.
+    /// @param data       Encoded update data.
+    event ConfigUpdate(UpdateType indexed updateType, bytes data);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Constructs the SuperchainConfig contract.
+    constructor() {
+        initialize({
+            _systemOwner: address(0),
+            _initiator: address(0),
+            _vetoer: address(0),
+            _guardian: address(0),
+            _delay: 0,
+            _sequencers: new Types.SequencerKeys[](0)
+        });
+    }
+
+    /// @notice Initializer.
+    ///         The resource config must be set before the require check.
+    /// @param _systemOwner     Initial owner of the contract.
+    /// @param _initiator Address of the initiator who may initiate an upgrade or change to critical config values.
+    /// @param _vetoer    Address of the vetoer.
+    /// @param _guardian  Address of the guardian, can pause the OptimismPortal.
+    /// @param _delay     The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    /// @param _sequencers The initial list of allowed sequencers
+    function initialize(
+        address _systemOwner,
+        address _initiator,
+        address _vetoer,
+        address _guardian,
+        uint256 _delay,
+        Types.SequencerKeys[] memory _sequencers
+    )
+        public
+        reinitializer(2)
+    {
+        _setAddress(SYSTEM_OWNER_SLOT, _systemOwner);
+        emit ConfigUpdate(UpdateType.SYSTEM_OWNER, abi.encode(_systemOwner));
+
+        _setAddress(INITIATOR_SLOT, _initiator);
+        emit ConfigUpdate(UpdateType.INITIATOR, abi.encode(_initiator));
```

Comment:
The `ConfigUpdate` event in the `ChainConfig` was meant to be consumed by the `op-node`. I don't know how useful the `[type, bytes]` style events are in this context

Reply:
yeah, I'm not attached to this, but I felt like there was a certain "Config Contract" pattern emerging from the `ProtocolVersion` and `SystemConfig` contracts. 

If that's not the case, then IMO we should just get rid of the whole eip1967 layout as well, it brings a ton of boilerplate with it.

Reply:
@tynes thinking about it more, I do like this format, whether or not the op-node reads it. 
Config contracts are inevitably going to get more and more values, having to add a new event for each one is onerous, and doesn't help off-chain code that much. 

Please LMK if you still wnat me to change it.